src/test/java/org/springframework/data/jpa/repository/UserCustomRepositoryFinderTests.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2008-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.jpa.repository;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.domain.sample.User;
+import org.springframework.data.jpa.repository.custom.UserCustomExtendedRepository;
+import org.springframework.data.repository.query.QueryLookupStrategy;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Integration test for executing custom finders, thus testing various query lookup strategies.
+ *
+ * @see QueryLookupStrategy
+ * @author Andriy Redko
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations = "classpath:config/namespace-customfactory-context.xml")
+@Transactional
+public class UserCustomRepositoryFinderTests {
```

Comment:
I don't think we need an Integration test for this.
Adding tests in `StringQueryUnitTests` should be sufficient.

Reply:
:+1: thanks!